contracts/logic/StakingPoolLogicV2.sol

```diff
@@ -63,7 +63,7 @@ library StakingPoolLogicV2 {
     StakingPoolV2.PoolData storage poolData,
     address user
   ) internal {
-    if(block.timestamp > poolData.endTimestamp) {
+    if(poolData.isOpened == true && block.timestamp > poolData.endTimestamp) {
```

Comment:
Please check the conditions under which calls the function.

Reply:
I think, this implementations can return invalid getRewardIndex. If there is no transactions like stake, `endTimestamp` is not up to date and `getRewardIndex()` return previous pool's reward index.